src/main/java/io/cdap/plugin/gcp/publisher/source/PubSubMessage.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.gcp.publisher.source;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.pubsub.v1.ReceivedMessage;
+
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Wrapper class for Pub/Sub message contents.
+ *
+ * We use this class to mirror the contents of Google's com.google.pubsub.v1.ReceivedMessage without having to add
+ * an additional exported dependency for our workers.
+ */
+public class PubSubMessage implements Externalizable {
+
+  private static final long serialVersionUID = 1L;
+
+  private String messageId;
+  private String orderingKey;
+  private String ackId;
+  private byte[] data;
+  private Map<String, String> attributes;
+  private Instant publishTime;
+
+  @VisibleForTesting
+  public PubSubMessage() {
+  }
+
+  @VisibleForTesting
+  public PubSubMessage(String messageId, String orderingKey, String ackId, byte[] data,
+                       Map<String, String> attributes, Instant publishTime) {
+    this.messageId = messageId;
+    this.orderingKey = orderingKey;
+    this.ackId = ackId;
+    this.data = data;
+    this.attributes = attributes;
+    this.publishTime = publishTime;
+  }
+
+  public PubSubMessage(ReceivedMessage message) {
+    if (message == null) {
+      return;
+    }
+
+    if (message.getMessage() != null) {
```

Comment:
There is too many calls to `message.getMessage()`, making it hard to read. Introduce a local variable.